http/src/request/channel/webhook/update_webhook_message.rs

```diff
@@ -254,20 +282,79 @@ impl<'a> UpdateWebhookMessage<'a> {
         Ok(self)
     }
 
-    fn request(&self) -> Result<Request> {
-        let body = crate::json_to_vec(&self.fields)?;
-        let route = Route::UpdateWebhookMessage {
-            message_id: self.message_id.0,
-            token: self.token.clone(),
-            webhook_id: self.webhook_id.0,
-        };
+    /// Attach a file to the webhook.
+    ///
+    /// This method is repeatable.
+    pub fn file(mut self, name: impl Into<String>, file: impl Into<Vec<u8>>) -> Self {
+        self.files.push((name.into(), file.into()));
+
+        self
+    }
+
+    /// Attach multiple files to the webhook.
+    pub fn files<N: Into<String>, F: Into<Vec<u8>>>(
+        mut self,
+        attachments: impl IntoIterator<Item = (N, F)>,
+    ) -> Self {
+        for (name, file) in attachments {
+            self = self.file(name, file);
+        }
 
-        Ok(if let Some(reason) = &self.reason {
-            let headers = request::audit_header(&reason)?;
-            Request::from((body, headers, route))
+        self
+    }
+
+    /// JSON encoded body of any additional request fields.
+    ///
+    /// If this method is called, all other fields are ignored, except for
```

Comment:
Why is this? The behavior of `payload_json` should be documented with an example.